x-pack/plugins/observability_solution/observability_ai_assistant/server/service/create_inference_endpoint.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import { Logger } from '@kbn/logging';
+
+export const AI_ASSISTANT_KB_INFERENCE_ID = 'ai_assistant_kb_inference';
+
+export async function createInferenceEndpoint({
+  esClient,
+  logger,
+}: {
+  esClient: {
+    asCurrentUser: ElasticsearchClient;
+  };
+  logger: Logger;
+}) {
+  try {
+    logger.debug(`Creating inference endpoint "${AI_ASSISTANT_KB_INFERENCE_ID}"`);
+    const response = await esClient.asCurrentUser.transport.request({
+      method: 'PUT',
+      path: `_inference/sparse_embedding/${AI_ASSISTANT_KB_INFERENCE_ID}`,
+      body: {
+        service: 'elser',
+        service_settings: {
+          num_allocations: 1,
+          num_threads: 1,
+        },
+      },
```

Comment:
For testing we should be able to deploy tiny_elser using the elasticsearch service (blocked by https://github.com/elastic/ml-team/issues/1298)

```
{
  method: 'PUT',
  path: `_inference/sparse_embedding/${AI_ASSISTANT_KB_INFERENCE_ID}`,
  body: {
    service: 'elasticsearch',
    service_settings: {
      num_allocations: 1,
      num_threads: 1,
      model_id: 'pt_tiny_elser',
    },
  },
}
```